ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts

```diff
@@ -27,20 +37,81 @@ export class HyperdriveJobComponent implements OnInit, OnDestroy {
   @Input() jobParameters: HyperdriveDefinitionParametersModel;
   @Output() jobParametersChange: EventEmitter<HyperdriveDefinitionParametersModel> = new EventEmitter();
   @Input() isJobTemplateSelected: boolean;
-  @Input() jobTemplateChanges: EventEmitter<string>;
+  @Input() jobTemplateChanges: EventEmitter<JobTemplateChangeEventModel>;
+  @Input() jobTemplateParameters: HyperdriveTemplateParametersModel;
+
+  private hyperdriveType: string = undefined;
 
   jobTemplateChangesSubscription: Subscription;
 
+  hyperdriveTypes = hyperdriveTypes;
+  hyperdriveTypesMap = hyperdriveTypesMap;
+  hyperdriveFields = hyperdriveFields;
+
   constructor() {
     // do nothing
   }
 
   ngOnInit(): void {
-    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value) => {
-      if (value) {
-        this.jobParametersChange.emit({ ...this.jobParameters, jobJar: undefined, mainClass: undefined });
+    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value: JobTemplateChangeEventModel) => {
+      this.jobParametersChange.emit({
+        ...this.jobParameters,
+        jobJar: undefined,
+        mainClass: undefined,
+        appArguments: this.getAppArguments(),
+      });
+      if (value?.jobTemplateId) {
+        const jobTemplateParameters = <HyperdriveTemplateParametersModel>value?.jobTemplateParameters;
+        this.hyperdriveType = HyperdriveUtil.getHyperdriveTypeFromAppArguments(
+          jobTemplateParameters.appArguments,
+          hyperdriveTypesJobTemplateFields,
+        );
       }
     });
+
+    if (this.isJobTemplateSelected) {
+      const hyperdriveTypeFromJobParams = HyperdriveUtil.getHyperdriveTypeFromAppArguments(
+        this.jobParameters.appArguments,
+        hyperdriveTypesFields,
+      );
+      const hyperdriveTypeFromTemplateParams = HyperdriveUtil.getHyperdriveTypeFromAppArguments(
+        this.jobTemplateParameters.appArguments,
+        hyperdriveTypesJobTemplateFields,
+      );
+
+      if (hyperdriveTypeFromJobParams == hyperdriveTypeFromTemplateParams) {
```

Comment:
Is this really necessary to check the hyperdrive type from `jobParameters.appArguments`? If yes, why do you not check it in the `jobTemplateChangesSubscription`?

Reply:
I think it is necessary to check the hyperdrive type, because for example if you are in show mode and the template would take precedence hyperdrive type would not be correct. I decided to show hyperdrive type only if it is 100% sure.

Reply:
Ok, but then I still think that the logic should be the same in the jobTemplateSubscription. You might also want to consider moving this logic to the util

Reply:
It is not really the same logic, it has to be executed also outside of the jobTemplateSubscription because logic insade is executed only on job template change so it would not get executed on init. 
If you remove job template you dont have to run logic of the finding correct hyperdrive type.

Move one part into util

Reply:
After discussion, I've refactored it